Storage/Processor/Processor.php

```diff
@@ -271,42 +271,26 @@ private function getResponseFromStreamInterface(StreamInterface $streamInterface
             $response->headers->set('Accept-Ranges', $request->isMethodSafe() ? 'bytes' : 'none');
         }
 
-        $range = $request->headers->get('Range');
-
-        if ($range === null) {
-            return $response;
-        }
-
-        list($start, $end) = explode('-', substr($range, 6), 2) + [0];
-
-        $end = ('' === $end) ? $fileSize - 1 : (int) $end;
-
-        if ('' === $start) {
-            $start = $fileSize - $end;
-            $end = $fileSize - 1;
-        } else {
-            $start = (int) $start;
-        }
-
-        if ($start > $end) {
+        $streamRange = new StreamRange($request->headers, $fileSize);
+        if ($streamRange->isOutOfRange()) {
             return $response;
         }
 
-        if ($start < 0 || $end > $fileSize - 1) {
+        if (!$streamRange->isSatisfiable()) {
             $response->setStatusCode(StreamedResponse::HTTP_REQUESTED_RANGE_NOT_SATISFIABLE);
-            $response->headers->set('Content-Range', sprintf('bytes */%s', $fileSize));
-        } elseif (0 !== $start || $end !== $fileSize - 1) {
-            $offset = $start;
+            $response->headers->set('Content-Range', $streamRange->getContentRangeHeader());
+        } elseif ($streamRange->rangeRequested()) {
             $response->setStatusCode(StreamedResponse::HTTP_PARTIAL_CONTENT);
-            $response->headers->set('Content-Range', sprintf('bytes %s-%s/%s', $start, $end, $fileSize));
-            $response->headers->set('Content-Length', strval($end - $start + 1));
+            $response->headers->set('Content-Range', $streamRange->getContentRangeHeader());
+            $response->headers->set('Content-Length', $streamRange->getContentLengthHeader());
 
-            $response->setCallback(function () use ($streamInterface, $offset, $end) {
```

Comment:
$streamInterface should be called "$stream"

Reply:
Done 9685e8cee31b7c945bdb7fd6b6ff2fb0bd475aa3